sentry-android-sqlite/src/test/java/io/sentry/android/sqlite/SentrySupportSQLiteStatementTest.kt

```diff
@@ -0,0 +1,158 @@
+package io.sentry.android.sqlite
+
+import androidx.sqlite.db.SupportSQLiteStatement
+import io.sentry.IHub
+import io.sentry.ISpan
+import io.sentry.SentryOptions
+import io.sentry.SentryTracer
+import io.sentry.SpanStatus
+import io.sentry.TransactionContext
+import org.mockito.kotlin.inOrder
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.whenever
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertNotNull
+import kotlin.test.assertTrue
+
+class SentrySupportSQLiteStatementTest {
+
+    class Fixture {
+        private val hub = mock<IHub>()
+        private val spanManager = SQLiteSpanManager(hub)
+        val mockStatement = mock<SupportSQLiteStatement>()
+        lateinit var sentryTracer: SentryTracer
+        lateinit var options: SentryOptions
+
+        fun getSut(sql: String, isSpanActive: Boolean = true): SentrySupportSQLiteStatement {
+            options = SentryOptions().apply {
+                dsn = "https://key@sentry.io/proj"
+            }
+            whenever(hub.options).thenReturn(options)
+            sentryTracer = SentryTracer(TransactionContext("name", "op"), hub)
+
+            if (isSpanActive) {
+                whenever(hub.span).thenReturn(sentryTracer)
+            }
+            return SentrySupportSQLiteStatement(mockStatement, spanManager, sql)
+        }
+    }
+
+    private val fixture = Fixture()
+
+    @Test
+    fun `all calls are propagated to the delegate`() {
+        val sql = "sql"
+        val statement = fixture.getSut(sql)
+
+        inOrder(fixture.mockStatement) {
+            statement.execute()
+            verify(fixture.mockStatement).execute()
+
+            statement.executeUpdateDelete()
+            verify(fixture.mockStatement).executeUpdateDelete()
+
+            statement.executeInsert()
+            verify(fixture.mockStatement).executeInsert()
+
+            statement.simpleQueryForLong()
+            verify(fixture.mockStatement).simpleQueryForLong()
+
+            statement.simpleQueryForString()
+            verify(fixture.mockStatement).simpleQueryForString()
+        }
+    }
+
+    @Test
+    fun `execute creates a span if a span is running`() {
+        val sql = "execute"
+        val sut = fixture.getSut(sql)
+        assertEquals(0, fixture.sentryTracer.children.size)
+        sut.execute()
+        val span = fixture.sentryTracer.children.firstOrNull()
+        assertSqlSpanCreated(sql, span)
+    }
+
+    @Test
+    fun `execute does not create a span if no span is running`() {
+        val sut = fixture.getSut("execute", isSpanActive = false)
+        sut.execute()
+        assertEquals(0, fixture.sentryTracer.children.size)
+    }
+
+    @Test
+    fun `executeUpdateDelete creates a span if a span is running`() {
+        val sql = "executeUpdateDelete"
+        val sut = fixture.getSut(sql)
+        assertEquals(0, fixture.sentryTracer.children.size)
+        sut.executeUpdateDelete()
```

Comment:
`l`: can we also add a test to verify the return value of this is what the delegate returns?